tests/test_extractor.py

```diff
@@ -1,6 +1,10 @@
 import matplotlib as mpl
 mpl.use('Agg')  # must be run first! But therefore requires noqa E402 on all other imports
 
+# This must be run before any astropy importing.
```

Comment:
Does this comment rely on pytest loading test files in the right order?

Reply:
First, it's not pytest here, it's nose.  Second, this would be a separate process, right?  So I think that's what matters.

Reply:
Ok. For pytest if you tell it to load 3 test files it will load them in the order you give to pytest and if the first test file imports astropy the comment in the second file about setting the env var won't do anything. In xdist mode you are at the mercy of the subprocess import order as well. Does astropy not have a way to programmatically add the cache dir?